atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -535,12 +538,23 @@ static TimestampCache timestampCache(
         LongSupplier cacheSize = () -> runtimeConfig.get().getTimestampCacheSize();
         Supplier<TimestampCache> timestampCacheSupplier = () ->
                 timestampStore.map(store ->
-                        OffHeapTimestampCache.create(store, metricsManager.getTaggedRegistry(), cacheSize))
+                        constructOffHeapTimestampCache(store, metricsManager.getTaggedRegistry(), cacheSize))
                         .orElseGet(() -> new DefaultTimestampCache(metricsManager.getRegistry(), cacheSize));
 
         return config.timestampCache().orElseGet(timestampCacheSupplier);
     }
 
+    private static TimestampCache constructOffHeapTimestampCache(
+                PhysicalPersistentStore persistentStore,
+                TaggedMetricRegistry taggedMetricRegistry,
+                LongSupplier cacheSize) {
+        OffHeapCache<Long, Long> offHeapCache = OffHeapCaches.create(
```

Comment:
This should live inside `OffHeapTimestampCache`. Otherwise it's easy to create something an offheap timestamp cache that can pass in different logical stores and what not. The abstraction is slowly falling apart here, but if you must have this, put it in `OffHeapTimestampCache`.